Mathlib/Data/Nat/Bitwise.lean

```diff
@@ -316,6 +319,14 @@ theorem testBit_two_pow (n m : ℕ) : testBit (2 ^ n) m = (n = m) := by
     simp [h]
 #align nat.test_bit_two_pow Nat.testBit_two_pow
 
+lemma and_two_pow (n i : ℕ) : n &&& 2 ^ i = (n.testBit i).toNat * 2 ^ i := by
```

Comment:
Can you add `two_pow_and` for symmetry? You should prove it via commutativity, not by copying the proof.

bors d+

Thanks!

Reply:
Added. If the CI goes through, I will push this PR.